services/user/Invite/src/AuthInvite.cpp

```diff
@@ -56,15 +56,18 @@ namespace UserService
       check(user == Invite::payerAccount, notWhitelisted.data());
    }
 
-   void AuthInvite::requireAuth(const PublicKey& pubkey)
+   void AuthInvite::requireAuth(const SystemService::AuthSig::SubjectPublicKeyInfo& pubkey)
    {
       auto claims = to<SystemService::Transact>().getTransaction().claims;
-      bool found  = std::find_if(claims.begin(), claims.end(),
-                                 [&](auto claim)
-                                 {
-                                   return claim.service == SystemService::VerifyK1::service &&
-                                          psio::from_frac<PublicKey>(claim.rawData) == pubkey;
-                                }) != claims.end();
+      bool found =
+          std::find_if(
+              claims.begin(), claims.end(),
+              [&](auto claim)
+              {
+                 std::vector<unsigned char> data{claim.rawData.begin(), claim.rawData.end()};
+                 return claim.service == SystemService::VerifySig::service &&
+                        SystemService::AuthSig::SubjectPublicKeyInfo{data} == pubkey;
```

Comment:
Unnecessary copy. You can just compare claim.rawData with pubkey.data (size + memcmp, since the they're different types).

Reply:
Fixed